pandas/core/generic.py

```diff
@@ -8564,6 +8523,246 @@ def nanptp(values, axis=0, skipna=True):
+
+    axis : {0 or 'index', 1 or 'columns'}, default 'index'
+        Axis to apply the function along (DataFrame only).
+
+        * 0 or 'index' : apply along the index, i.e. the function
+          gets each column.
+        * 1 or 'columns' : apply along the columns, i.e. the function
+          gets each row.
+
+    *args
+        Additional positional arguments are passed to `func`.
+    **kwargs
+        Additional keyword arguments are passed to `func`.
+
+    Returns
+    -------
+    transformed : Series or DataFrame
+        The result type is the same as the input. Furthermore, it is
+        the same shape.
+
+    Notes
+    ------
+    The function performs the transformation.
+    It takes a Series as input and must return a same length Series.
+    Note that the function may modify the Series index.
+
+    If applied to a Series the function is applied once.
+    If applied to a DataFrame the function is applied on each row or column.
+    Note that if the function modifies the index it must do so consistently
+    in the case of a DataFrame.
+
+    See also
+    --------
+    pandas.DataFrame.aggregate : Perform aggregate type of operations.
+    pandas.DataFrame.apply : Apply general functions.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame(np.arange(30).reshape(10, 3),
+    ...                   columns=['A', 'B', 'C'],
+    ...                   index=pd.date_range('1/1/2000', periods=10))
+    >>> df.iloc[3:7] = np.nan
+    >>> df
+                A     B     C
+    2000-01-01   0.0   1.0   2.0
+    2000-01-02   3.0   4.0   5.0
+    2000-01-03   6.0   7.0   8.0
+    2000-01-04   NaN   NaN   NaN
+    2000-01-05   NaN   NaN   NaN
+    2000-01-06   NaN   NaN   NaN
+    2000-01-07   NaN   NaN   NaN
+    2000-01-08  21.0  22.0  23.0
+    2000-01-09  24.0  25.0  26.0
+    2000-01-10  27.0  28.0  29.0
+
+    By default, `func` is applied to each column independently.
+
+    >>> df.transform(lambda x: (x - x.mean()) / x.std())
+                A         B         C
+    2000-01-01 -1.143001 -1.143001 -1.143001
+    2000-01-02 -0.889001 -0.889001 -0.889001
+    2000-01-03 -0.635001 -0.635001 -0.635001
+    2000-01-04       NaN       NaN       NaN
+    2000-01-05       NaN       NaN       NaN
+    2000-01-06       NaN       NaN       NaN
+    2000-01-07       NaN       NaN       NaN
+    2000-01-08  0.635001  0.635001  0.635001
+    2000-01-09  0.889001  0.889001  0.889001
+    2000-01-10  1.143001  1.143001  1.143001
+
+    Apply `func` to each row with ``axis='columns'``.
+
+    >>> df.transform(lambda x: (x - x.mean()) / x.std(), axis='columns')
+                A    B    C
+    2000-01-01 -1.0  0.0  1.0
+    2000-01-02 -1.0  0.0  1.0
+    2000-01-03 -1.0  0.0  1.0
+    2000-01-04  NaN  NaN  NaN
+    2000-01-05  NaN  NaN  NaN
+    2000-01-06  NaN  NaN  NaN
+    2000-01-07  NaN  NaN  NaN
+    2000-01-08 -1.0  0.0  1.0
+    2000-01-09 -1.0  0.0  1.0
+    2000-01-10 -1.0  0.0  1.0
+
+    >>> def f(s):
```

Comment:
Can you explain the motivation for this example? What's it showing that the previous one didn't?

Reply:
It shows that the input to the function is a pandas.Series, as opposed to an unspecified kind of vector. I remember this is something that was not clear to me when I started using pandas. So I figured it might help others too.

Reply:
Yes, that's good (though `ndarray` has size as well). Perhaps something like

```
In [7]: def cell_name(series):
   ...:     return ['{}-{}'.format(series.name, idx) for idx in series.index]
   ...:

In [8]: df = pd.DataFrame({"A": [10, 20, 30], "B": [1, 2, 3]})

In [9]: df.transform(cn)
Out[9]:
     A    B
0  A-0  B-0
1  A-1  B-1
2  A-2  B-2
```

Reply:
I put your example instead of mine.  
It is undoubtedly better.